libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/ListItem.kt

```diff
@@ -0,0 +1,419 @@
+import androidx.compose.ui.unit.Dp
+import io.element.android.libraries.designsystem.preview.ElementThemedPreview
+import io.element.android.libraries.designsystem.preview.PreviewGroup
+import io.element.android.libraries.theme.ElementTheme
+
+// Designs: https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=425%3A24208&mode=design&t=G5hCfkLB6GgXDuWe-1
+
+/**
+ * A List Item component to be used in lists and menus with simple layouts, matching the Material 3 guidelines.
+ * @param headlineContent The main content of the list item, usually a text.
+ * @param modifier The modifier to be applied to the list item.
+ * @param overlineContent The content to be displayed above the headline content.
+ * @param supportingContent The content to be displayed below the headline content.
+ * @param leadingContent The content to be displayed before the headline content.
+ * @param trailingContent The content to be displayed after the headline content.
+ * @param colors The colors to be used for the list item.
+ * @param tonalElevation The tonal elevation to be used for the list item.
+ * @param shadowElevation The shadow elevation to be used for the list item.
+ * @param isPrimaryAction Whether the list item is a primary action. This will change the color of the leading content to use primary tokens.
+ * @param isDestructive Whether the list item is a destructive action. This will change the color of the headline content to use critical tokens.
+ * @param enabled Whether the list item is enabled. When disabled, will change the color of the headline content and the leading content to use disabled tokens.
+ * @param onClick The callback to be called when the list item is clicked.
+ *
+ * **Note:** [isPrimaryAction], [isDestructive] and [enabled] should only be used in single line [ListItem] components,
+ * and only one should be customised at a time.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun ListItem(
+    headlineContent: @Composable () -> Unit,
+    modifier: Modifier = Modifier,
+    overlineContent: @Composable (() -> Unit)? = null,
+    supportingContent: @Composable (() -> Unit)? = null,
+    leadingContent: @Composable (() -> Unit)? = null,
+    trailingContent: @Composable (() -> Unit)? = null,
+    colors: ListItemColors = ListItemDefaults.colors(),
+    tonalElevation: Dp = ListItemDefaults.Elevation,
+    shadowElevation: Dp = ListItemDefaults.Elevation,
+    isPrimaryAction: Boolean = false,
+    isDestructive: Boolean = false,
+    enabled: Boolean = true,
+    onClick: () -> Unit = { },
+) {
+    val headlineColor = when {
+        isDestructive -> ElementTheme.colors.textCriticalPrimary
+        !enabled -> ElementTheme.colors.textDisabled
+        else -> ElementTheme.colors.textPrimary
+    }
+    val leadingContentColor = when {
+        isPrimaryAction -> ElementTheme.colors.iconPrimary
+        isDestructive -> ElementTheme.colors.iconCriticalPrimary
```

Comment:
Maybe test `isDestructive` first?

After thinking about it, I understand that parameters `isPrimaryAction` and `isDestructive` are probably exclusive (i.e. they cannot be both equal to `true`), but maybe enforce this by introducing an enum instead? If I am wrong, maybe add a preview with both parameter to `true`?

Reply:
I implemented a `style` parameter to handle this in an enum way. Thanks!